api/src/modules/shared/services/slot.service.ts

```diff
@@ -114,6 +114,31 @@ export class SlotService {
 		});
 	}
 
+	private async checkSlotLength(slot: Partial<Slot>) {
+		const slotLength = (slot.end - slot.start) / 60;
+		const tenant = this.tenantService.findOne();
+		const minLength = (await tenant).settings.minimumSlotDuration || 30
+		const maxLength = (await tenant).settings.maximumSlotDuration || 1440
```

Comment:
To improve performance you could await on line 119 so you dont have to fetch the tenant data twice

Reply:
Except for that, looks gud!